app/models/user.rb

```diff
@@ -36,39 +42,37 @@ def liked?(post)
   def name
     "#{firstname} #{surname}"
   end
+  
   def friends
-    friends_array = active_friendships.map { |friendship| friendship.active_friend if friendship.confirmed }
-    friends_array.concat(passive_friendships.map { |friendship| friendship.passive_friend if friendship.confirmed })
-    friends_array.compact
+    active_friends.compact
   end
 
-  def friends_id
-    friends_array = active_friendships.map { |friendship| friendship.active_friend.id }
-    friends_array.concat(passive_friendships.map { |friendship| friendship.passive_friend.id})
-    friends_array.compact
+  def friends_and_requests_id
+    all_friends_and_requests = active_friends.ids.compact.concat(incoming_pending_friends.ids.compact)
+    all_friends_and_requests += all_friends_and_requests.concat(outgoing_pending_friends.ids.compact)
   end
 
-  def pending_requests_to_users
-    active_friendships.map { |friendship| friendship.active_friend if !friendship.confirmed }.compact
-  end
+  # def pending_requests_to_users
+  #   active_friendships.map { |friendship| friendship.active_friend if !friendship.confirmed }.compact
+  # end
 
-  def pending_requests_from_users
-    passive_friendships.map { |friendship| friendship.passive_friend if !friendship.confirmed }.compact
-  end
+  # def pending_requests_from_users
+  #   passive_friendships.map { |friendship| friendship.passive_friend if !friendship.confirmed }.compact
+  # end
```

Comment:
**[REQUIRED]**
Let's remove the commented code. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from the Pathwright).